### PR TITLE
Hardened used pathes for #include preprocessor in scala files

### PR DESCRIPTION
**Pull request by lefou**
By using Twitters eval util, we also derive it's `#include` preprocessor, which by default is a bit unexact when it comes to potential source locations for includable files. Also the include path is unstable when Maven is run in (multi-project) reactor builds.

This patch removes the ambiguities and also makes include pathes more stable when run in reactor builds.

### CODE REVIEW

**Comment by lefou** [comment]
Any coments or questions about this? @huntc, maybe?

**Comment by lefou** [comment]
Can anyone review and/or comment, please?

**Comment by jvanzyl** [comment]
If you are doing the work and you're learning how to work on this implementation feel free to merge. It most of the other folks here don't know much about Scala. Happy to give you access if you want to take care of it yourself.

**Comment by huntc** [comment]
Sorry, I should have picked this up also. I'll get to it in the next 24h.

**Comment by lefou** [comment]
Great, to both of you! I'm waiting for @huntc's review and hopefully merge then. But merging by myself would be fine too.

**Comment by huntc** [comment]
Would it be possible to have a unit test for the new functionality?

File: polyglot-scala/src/main/scala/org/sonatype/maven/polyglot/scala/ScalaModelReader.scala [hunk 1]
```diff
@@ -168,8 +170,33 @@ class ScalaModelReader @Inject()(executeManager: ExecuteManager) extends ModelRe
...
+
+  private def eval(evalPomFile: File, sourcePomFile: File, options: util.Map[String, _]): ScalaModel = {
+    val sourceFile = new File(PolyglotModelUtil.getLocation(options))
+    val eval = new MvnEval(Some(evalPomFile.getParentFile), sourceFile.getParentFile())
+    try {
+      eval.apply[ScalaModel](sourcePomFile)
+    } catch {
+      case e: eval.CompilerException =>
+        // TODO: parse line nr etc
+        throw new ModelParseException("Cannot compile pom file: " + sourceFile, 0, 0, e)
```
Comment (huntc) [inline polyglot-scala/src/main/scala/org/sonatype/maven/polyglot/scala/ScalaModelReader.scala]: Can we resolve the todo?

File: polyglot-scala/src/main/scala/org/sonatype/maven/polyglot/scala/ScalaModelReader.scala [hunk 2]
```diff
@@ -168,8 +170,33 @@ class ScalaModelReader @Inject()(executeManager: ExecuteManager) extends ModelRe
     new File(evalTarget, "pom.scala")
   }
 
-  private def eval(evalPomFile: File, sourcePomFile: File): ScalaModel = {
-    new Eval(Some(evalPomFile.getParentFile)).apply[ScalaModel](sourcePomFile)
+  class MvnEval(target: Option[File], includeBaseDir: File) extends Eval(target) {
```
Comment (huntc) [inline polyglot-scala/src/main/scala/org/sonatype/maven/polyglot/scala/ScalaModelReader.scala]: Can we have API doc for new types describing their motivation?

**Comment by lefou** [comment]
@huntc, thanks for your review.

Unit testing this PR is quite hard. Exspecially, as the unit under test would be the complete JVM. The issues we try to solve here depend on the runtime environment and configuration of the JVM. In detail:

* The current working directory of the JVM process
* The classpath of the JVM
* The state and content of the local filesystem

In other words, we essentially need to test a Java / Maven process in action. 

Are there any integration tests for polyglot-maven? If so, that would be the best place to add a test case.

I will target your other review comments in an additional commit.

**Comment by mkristian** [comment]
@lefou the ruby DSL part has some integration tests

**Comment by lefou** [comment]
Can we merge, now? I do not plan to create a integration test as part of this PR. This PR is essentially only the missing configuration for a features which is already present but unusable.

**Comment by huntc** [comment]
Thanks!

**Comment by lefou** [comment]
Thank you!

**Comment by lefou** [comment]
Could we have a release with that feature included, please?

**Comment by jvanzyl** [comment]
Sure

**Comment by lefou** [comment]
Great! Looking forward to it. Thanks.

**Comment by lefou** [comment]
What's the planned release date?

**Comment by jvanzyl** [comment]
It should be in Maven Central shortly.

**Pull request merged**